### PR TITLE
Add MediaTracker playback scenario test cases

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Add MediaTracker playback scenario test cases. Tests flow between MediaTrackerEventGenerator and MediaEventTracker for various playback and ad scenarios.

Test scenarios were ported from iOS functional test scenarios found here: https://github.com/adobe/aepsdk-edgemedia-ios/tree/dev/Tests/FunctionalTests/Scenarios
All iOS test scenarios were ported without any changes to test expectations.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java [hunk 1]
```diff
@@ -40,14 +40,22 @@ public class MediaPublicTracker implements MediaTracker {
     private Map<String, Object> lastPlayheadParams;
 
     @VisibleForTesting
-    MediaPublicTracker(final String trackerId, final AdobeCallback<Event> eventConsumer) {
+    interface TimestampSupplier {
```
Comment (addb) [inline code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java:43]: nit: TimestampProvider

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java [hunk 2]
```diff
@@ -40,14 +40,22 @@ public class MediaPublicTracker implements MediaTracker {
...
+        long getCurrentTimestamp();
+    }
+
+    @VisibleForTesting
+    TimestampSupplier currentTimestamp = () -> Calendar.getInstance().getTimeInMillis();
```
Comment (addb) [inline code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java:48]: nit: TimeStampProvider timeStampProvider

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java [hunk 3]
```diff
@@ -205,18 +213,19 @@ synchronized void trackInternal(
...
                 && params != null) {
             lastPlayheadParams = new HashMap<>(params);
         }
+
+        eventConsumer.call(event);
     }
 
     long getCurrentTimestamp() {
-        return Calendar.getInstance().getTimeInMillis();
+        return currentTimestamp.getCurrentTimestamp();
```
Comment (addb) [inline code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java:228]: timeStampProvider.getCurrentTimestamp

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java (hunk 1 above)
Comment (kevinlind) [reply code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java:43]: I named the interface similar to Java's Functional Interfaces, https://docs.oracle.com/javase/8/docs/api/java/util/function/package-summary.html
Java provides a set of interfaces which may be used as targets for Lambda expressions. For example LongConsumer is an interface which takes a Long and returns no result, and LongSupplier is an interface which takes nothing and returns a Long. Unfortunately the minimum required Android API to use the functional interfaces is 24, which is why I created the TimestampSupplier.

File: code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java [hunk 4]
```diff
@@ -195,7 +203,7 @@ synchronized void trackInternal(
             eventData.put(MediaInternalConstants.EventDataKeys.Tracker.EVENT_METADATA, metadata);
         }
 
-        long ts = getCurrentTimestamp();
+        long ts = getTimestampSupplier();
```
[the remark this answers is not in the document]
Comment (kevinlind) [reply code/edgemedia/src/phone/java/com/adobe/marketing/mobile/edge/media/internal/MediaTrackerEventGenerator.java]: Sorry, I didn't mean to rename that function. I reverted it back to `getCurrentTimestamp()`.

**Review by addb — APPROVED** [review]

**Pull request merged**